components/board/interrupt_stream.go

```diff
@@ -75,13 +73,8 @@ func (s *interruptStream) startStream(ctx context.Context, interrupts []DigitalI
 	}
 }
 
-func (s *interruptStream) recieveFromStream(ctx context.Context, stream pb.BoardService_StreamTicksClient, ch chan Tick) {
-	defer func() {
-		s.streamMu.Lock()
-		defer s.streamMu.Unlock()
-		s.streamRunning = false
-	}()
-	// Close the stream ready channel so the above function returns.
+func (s *interruptStream) receiveFromStream(ctx context.Context, stream pb.BoardService_StreamTicksClient, ch chan Tick) {
+	// Close the stream ready channel so startStream returns.
 	if s.streamReady != nil {
```

Comment:
im having trouble remembering why we need the streamReady - do we need to delay the return of `startStream`  to keep this function's context from canceling?

Reply:
:raised_eyebrow: You make a good point: it's not clear to me why `startStream` needs to wait for the background goroutine to start at all. Perhaps we can get rid of that altogether. I'll think some more...